File: frontend/src/gql/mutations/rating.ts
import { gql } from '@apollo/client';

export const ADD_RATING = gql`
  mutation AddRating($input: RatingInput!) {
    addRating(rating: $input) {
      id
      matchPlayerId
      score
      userId
      createdAt
      matchPlayer {
        id
        playerId
        lineupId
        teamId
        position
        isStarter
        averageRating
        ratings {
          id
          score
          userId
        }
      }
    }
  }
`;

export const ADD_SIMPLE_RATING = gql`
  mutation AddSimpleRating($matchPlayerId: String!, $score: Int!) {
    addSimpleRating(matchPlayerId: $matchPlayerId, score: $score) {
      id
      matchPlayerId
      score
      userId
      createdAt
      matchPlayer {
        id
        playerId
        lineupId
        teamId
        position
        isStarter
        averageRating
        ratings {
          id
          score
          userId
          createdAt
        }
      }
    }
  }
`;

File: backend/src/Norskball/Models/Rating.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Norskball.Models;

public class Rating
{
    public string Id { get; set; } = default!;
    public string MatchPlayerId { get; set; } = default!;
    public string UserId { get; set; } = default!;
    public int Score { get; set; }  // 1-10
    public DateTime CreatedAt { get; set; }

    // Navigation property
    [ForeignKey("MatchPlayerId")]
    public MatchPlayer MatchPlayer { get; set; } = default!;
} 